### PR TITLE
Sparse

### PR DESCRIPTION
**Pull request by fieker**

### CODE REVIEW

File: src/Sparse/HNF.jl [hunk 1]
```diff
@@ -478,7 +485,8 @@ function hnf_extend!(A::SMat{fmpz}, b::SMat{fmpz}, trafo::Type{Val{N}} = Val{fal
       if nc % 10 == 0
         println("Now at $nc rows of $(nrows(b)), HNF so far $(nrows(A)) rows")
         println("Current density: $(density(A))")
-        println("and size of largest entry: $(nbits(maximum(abs, A))) bits")
+        println("and size of largest entry: $(nbits(maximum(abs, A))) bits $(sum(nbits, A))")
+        println("#fmpz: $(Nemo.no_fmpz), ", unsafe_load(cglobal((:mpz_free_alloc, Nemo.libflint), Cint)), ", ", unsafe_load(cglobal((:mpz_free_num, Nemo.libflint), Cint)))
```
Comment (thofma) [inline src/Sparse/HNF.jl:489]: I guess we have to remove this.

Comment (thofma) [inline src/Sparse/HNF.jl:489]:
```suggestion
```

Comment (fieker) [reply src/Sparse/HNF.jl:489]: indeed, sorry

**Comment by thofma** [comment]
Tests seem to be failing with a real error.

**Pull request merged**